ops/data_repo_terra/import_snapshot_data.py

```diff
@@ -0,0 +1,92 @@
+    client = bigquery.Client(project=google_project, credentials=credentials)
+    query = f'SELECT * FROM `{google_project}.{datarepo_snapshot}.sample`'
+    query_job = client.query(query)
+    return query_job.result()
+
+def format_data_as_tsv(tsv_file, snapshot_rows, data_table):
+    """ Write BigQuery results into a TSV file for importing to Terra.
+    The first row header must follow the format 'entity:{data_table}_id'.
+    For example, 'entity:sample_id' will upload the tsv data into a "sample" table in
+    the workspace (or create one if it does not exist). If the table already contains
+    a sample with that id, it will get overwritten."""
+    entities = []
+    with open(tsv_file, 'w') as f:
+        headers = None
+        writer = csv.writer(f, delimiter='\t')
+        for row in snapshot_rows:
+            if not headers:
+                keys = list(row.keys())
+                keys[0] = f'entity:{data_table}_id'
+                writer.writerow(keys)
+            writer.writerow(row.values())
+            entities.append({'entity_name': data_table, "entity_id": row[0]})
+    return entities
+
+def upload_to_terra(terra_url, terra_workspace, tsv_file, service_account_path):
+    """Upload a TSV file containing sample inputs to a terra workspace. The service
+    account must have owner permissions on the workspace."""
+    #TODO: Find out what the file size/upload limitations are
+    import_url = f'{terra_url}/api/workspaces/{terra_workspace}/flexibleImportEntities'
+    scopes = ['email', 'openid', 'profile']
+    credentials = get_service_account_credentials(service_account_path, scopes)
+    headers = {'Authorization': f'Bearer {credentials.token}'}
+    with open (tsv_file, 'rb') as f:
+        contents = f.read()
+    response = requests.post(import_url,
+                             headers=headers,
+                             files={'entities': io.BytesIO(contents),
+                                    'type': 'text/tab-separated-values'})
+    return response
+
+def main(datarepo_snapshot, terra_url, terra_workspace, terra_data_table, service_account_path):
+    tsv_file = f'{uuid.uuid4()}_samples.tsv'
+    snapshot_rows = get_snapshot_data('broad-jade-dev-data', datarepo_snapshot, service_account_path)
+    entities = format_data_as_tsv(tsv_file, snapshot_rows, terra_data_table)
+    upload_to_terra(terra_url, terra_workspace, tsv_file, service_account_path)
+    print(f'The following samples have been uploaded to {terra_workspace}:')
+    for sample in entities:
+        print(sample)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Import Data Repo snapshot data into a Terra workspace.",
+                                     usage="%(prog)s [-h] [DATAREPO_SNAPSHOT TERRA_URL TERRA_WORKSPACE SERVICE_ACCOUNT_PATH] [...]")
+    parser.add_argument("--datarepo_snapshot",
+                        default="zerosnapshot",
```

Comment:
should it be `wflsnapshot`, or zero means a different thing here?

Reply:
The original dataset had `zero` in it instead of `wfl` so I used it here for consistency!